docs/getting_started/DMUpgrade.md

```diff
@@ -0,0 +1,313 @@
+
+## How to perform force upgrade at runtime?
+1. Ensure there are no CR objects for the dsl node crd that is being upgraded
+Port forward port 5001 of nexusapp-mgr pod in cosmos-cd namespace
+```
+kubectl port-forward -n cosmos-cd nexus-app-mgr-7865f8df58-h7hbj 5018:5001
+```
+2. Invoke the tenant/upgrade POST end point by providing the TenantID and Force info in payload<br>
+Example:<br>
+HTTP METHOD: POST<br>
+URL: http://localhost:5018/tenant/upgrade <br>
+Payload:
+```
+{
+    "TenantID": "v4",
+    "DatamodelForceInstall": true
+}
+```
+## What is considered a non-breaking change?
+
+#####  1. Add an optional field in the nexus node (field with "omitempty" tag )
+
+<details>
+<summary>Show example</summary>
+
+1. Add an optional field called `Location` (field with `omitempty` tag) in the node
+    ```
+    type Leader struct {
+        // Tags "Root" as a node in datamodel graph
+        nexus.Node
+
+       Name          string
+       Designation   string
+       DirectReports Manager `nexus:"children"`
+    +  Location      string  `json:"location,omitempty"`
+    }
+
+2. Rebuild your datamodel
+   ```
+   nexus datamodel build --name orgchart
+   ```
+
+   Now, the build would succeed
+
+</details>
+
+#####  2. Modify the REST APIs URIs
+
+<details>
+<summary>Show example</summary>
+
+1. Remove the GET URI from the spec
+    ```shell
+   var LeaderRestAPISpec = nexus.RestAPISpec{
+    Uris: []nexus.RestURIs{
+    -   {
+    -       Uri:     "/leader/{root.Leader}",
+    -       Methods: nexus.DefaultHTTPMethodsResponses,
+    -   },
+        {
+            Uri:     "/leaders",
+            Methods: nexus.HTTPListResponse,
+        },
+    },
+   }
+
+   // nexus-rest-api-gen:LeaderRestAPISpec
+   type Leader struct {
+      nexus.Node    Name          string
+      Designation   string
+      DirectReports Manager `nexus:"children"`
+      Location      string  `json:"location,omitempty"`
+   }
+
+2. Rebuild your datamodel
+   ```
+   nexus datamodel build --name orgchart 
+   ```
+
+   Now, the build would succeed
+
+</details>
+
+#####  3. Modify the REST APIs annotation
+
+<details>
+<summary>Show example</summary>
+
+1. Modify the `nexus-rest-api-gen` annotation spec from `LeaderRestAPISpec` to `NewLeaderRestAPISpec`
+
+   ```shell
+   var NewLeaderRestAPISpec = nexus.RestAPISpec{
+    Uris: []nexus.RestURIs{
+        {
+            Uri:     "/leaders",
+            Methods: nexus.HTTPListResponse,
+        },
+    },
+   }
+
+   // nexus-rest-api-gen:NewLeaderRestAPISpec <==
+   type Leader struct {    
+      nexus.Node    Name          string
+      Designation   string
+      DirectReports Manager `nexus:"children"`
+      Location      string  `json:"location,omitempty"`
+   }
+
+2. Rebuild your datamodel
+   ```
+   nexus datamodel build --name orgchart 
+   ```
+
+   Now, the build would succeed
+
+</details>
+
+## What is considered a backward incompatible change?
+
```

Comment:
We also need examples for changes to RestAPISpec that are flagged as backward incompatible.